tests/test_os_sorted.py

```diff
@@ -0,0 +1,200 @@
+        "/p/Folder/file.x1.9.tar.gz",
+        "/p/Folder (2)/file.tar.gz",
+        "/p/Folder (1)/file.tar.gz",
+        "/p/Folder/file.x1.10.tar.gz",
+    ]
+    expected = [
+        "/p/Folder/file.x1.9.tar.gz",
+        "/p/Folder/file.x1.10.tar.gz",
+        "/p/Folder (1)/file.tar.gz",
+        "/p/Folder (1)/file (1).tar.gz",
+        "/p/Folder (2)/file.tar.gz",
+        "/p/Folder (10)/file.tar.gz",
+    ]
+    result = natsort.os_sorted(given)
+    assert result == expected
+
+
+def test_os_sorted_misc_no_fail():
+    natsort.os_sorted([9, 4.3, None, float("nan")])
+
+
+def test_os_sorted_key():
+    given = expected = ["foo0", "goo1", "foo2"]
+    result = natsort.os_sorted(given, key=lambda x: x.replace("g", "f"))
+    assert result == expected
+
+
+# The following is a master list of things that might give trouble
+# when sorting like the file explorer.
+given = [
+    "11111",
+    "!",
+    "#",
+    "$",
+    "%",
+    "&",
+    "'",
+    "(",
+    ")",
+    "+",
+    "+11111",
+    "+aaaaa",
+    ",",
+    "-",
+    ";",
+    "=",
+    "@",
+    "[",
+    "]",
+    "^",
+    "_",
+    "`",
+    "aaaaa",
+    "foo0",
+    "foo_0",
+    "{",
+    "}",
+    "~",
+    "§",
+    "°",
+    "´",
+    "µ",
+    "€",
+    "foo1",
+    "foo2",
+    "foo4",
+    "foo10",
+    "Foo3",
+]
+
+# The expceted values change based on the environment
+if platform.system() == "Windows":
+    expected = [
+        "'",
+        "-",
+        "!",
+        "#",
+        "$",
+        "%",
+        "&",
+        "(",
+        ")",
+        ",",
+        ";",
+        "@",
+        "[",
+        "]",
+        "^",
+        "_",
+        "`",
+        "{",
+        "}",
+        "~",
+        "´",
+        "€",
+        "+",
+        "+11111",
+        "+aaaaa",
+        "=",
+        "§",
+        "°",
+        "µ",
+        "11111",
+        "aaaaa",
+        "foo_0",
+        "foo0",
+        "foo1",
+        "foo2",
+        "Foo3",
+        "foo4",
+        "foo10",
+    ]
+
+elif has_icu:
+    expected = [
+        "_",
+        "-",
+        ",",
+        ";",
+        "!",
+        "'",
+        "(",
+        ")",
+        "[",
+        "]",
+        "{",
+        "}",
+        "§",
+        "@",
+        "&",
+        "#",
+        "%",
+        "`",
+        "´",
+        "^",
+        "°",
+        "+",
+        "+11111",
+        "+aaaaa",
+        "=",
+        "~",
+        "$",
+        "€",
+        "11111",
+        "aaaaa",
+        "foo_0",
+        "foo0",
+        "foo1",
+        "foo2",
+        "Foo3",
+        "foo4",
+        "foo10",
+        "µ",
+    ]
+else:
+    # For non-ICU UNIX, the order is all over the place
+    # from platform to platform, distribution to distribution.
+    # It's not really possible to predict the order across all
+    # the different OS. To work around this, we will exclude
+    # the special characters from the sort.
+    given = given[0:1] + given[22:25] + given[33:]
```

Comment:
maybe you should lump them all together into a separate list (`given_special`), and append them in the cases above, instead.  seems easy to make a mistake, and have it pass on some os'es but not others.

Reply:
That is of course a much better solution.  When I wrote that I knew it was not good design... but I was tired and frustrated that things weren't working as smoothly as I wanted.

Reply:
Fixed!